Fix KeyError: 'padding' in _avg_pool2d implementation

Thrsu opened the pull request:
This PR fixes a bug in the avg_pool2d implementation. The bug causes a KeyError when running the provided code snippet. The error message is as below:
```
Traceback (most recent call last):
    ...
    mod = from_fx(fx_model, input_info)
  File "/workplace/software/tvm/tvm/python/tvm/relax/frontend/torch/fx_translator.py", line 1449, in from_fx
    return TorchFXImporter().from_fx(
  File "/workplace/software/tvm/tvm/python/tvm/relax/frontend/torch/fx_translator.py", line 1336, in from_fx
    self.env[node] = self.convert_map[func_name](node)
  File "/workplace/software/tvm/tvm/python/tvm/relax/frontend/torch/fx_translator.py", line 763, in _avg_pool2d
    stride = node.args[2] if nargs > 2 else node.kwargs["stride"]
KeyError: 'stride'
```
And here is the code to reproduce this bug.
```python
import torch
from torch import fx
from torch.nn import Module
import tvm
from tvm import relax
from tvm.relax.frontend.torch import from_fx

input_data = torch.randn([1, 1, 3, 3], dtype=torch.float32)
para_1 = (2, 1)
class avg_pool2d(Module):
    def forward(self, input):
        return torch.nn.functional.avg_pool2d(input, para_1,divisor_override=2,)
model = avg_pool2d().float()
input_data = [input_data]
input_info = list(zip([list(inp.shape) for inp in input_data], [str(inp.dtype) for inp in input_data]))
fx_model : torch.fx.GraphModule = fx.symbolic_trace(model)
with torch.no_grad():
    mod = from_fx(fx_model, input_info)
```
The issue arises due to the lack of a check for the existence of the "stride" key in the code. To resolve this bug, I have modified the code to include a check for the existence of the "stride" key before accessing it. 

I have tested these changes by running the provided code snippet, and the KeyError is no longer thrown. The fix ensures that the code gracefully handles cases where the "stride" key is missing.

Please review and consider merging this PR. Thank you! 
cc @jikechao @Hzfengsy @leandron

jikechao commented:
@Thrsu Thanks for the PR.
Please change the title to `[Unity][Bugfix]Fix KeyError: 'padding' in _avg_pool2d implementation` and add a test case.

jikechao commented:
@tvm-bot rerun

Hzfengsy commented:
Would be great if you could add a regression test.

BTW, tvm bot does not work on unity branch :(

Thrsu commented:
> @Thrsu Thanks for the PR.
> Please change the title to `[Unity][Bugfix]Fix KeyError: 'padding' in _avg_pool2d implementation` and add a test case.
Thank you for your review, I have changed the title and added a test case, plz recheck.

Thrsu commented:
> Would be great if you could add a regression test.
> 
> BTW, tvm bot does not work on unity branch :(

Sure, I have added a regression test as requested. Please recheck.

jikechao reviewed (approved):
LGTM, Thanks for your efforts.

Hzfengsy commented on src/tir/transforms/lower_thread_allreduce.cc:
```diff
@@ -429,6 +429,7 @@ class ThreadAllreduceBuilder final : public StmtExprMutator {
     // Fix all local allocations as all statements are built.
     Stmt body = SeqStmt::Flatten(seq);
     for (Buffer buf : new_alloc_bufs) {
+      body = DeclBuffer(buf, body);
```
I wonder about the changes here. It looks like not related to this PR.

Hzfengsy commented on tests/python/unittest/test_tir_transform_lower_thread_all_reduce.py:
ditto

Thrsu replied:
I apologize for any confusion caused by the unrelated changes in this pull request. It appears that when I submitted the PR, I mistakenly synchronized the commit history from the original repository, which inadvertently included modifications from other pull requests. However, the actual changes made in this PR are only related to the fx_translator.py and test_frontend_from_fx.py files.

Hzfengsy replied:
Could you please rebase it to the lastest commit?

Thrsu replied:
I have rebased it, please recheck.
